pkg/controller/certificates/issuing/issuing_controller_test.go

```diff
@@ -71,6 +76,8 @@ func TestIssuingController(t *testing.T) {
 		gen.SetCertificateDNSNames("example.com"),
 		gen.SetCertificateRevision(1),
 		gen.SetCertificateNextPrivateKeySecretName(nextPrivateKeySecretName),
+		gen.SetCertificateCommonName("commonname"),
+		gen.SetCertificateExtraName(givenName),
```

Comment:
So... I thought with my change here I was testing that my change was working and adding these fields, but writing more tests elsewhere in the code (ca_test.go), now I'm not so sure. 

Taking a hard look, I don't actually understand what maps the OIDs into the certificate so that something knows when it sees "2.4.5.43" or whatever it should display it as "GN" without some sort of actual mapping. I had assumed it was some algorithm based off the number array, but now I no longer believe that to be the case.

I was looking. at the other PR you had mentioned and it's trying to achieve something very similar and I see it is doing a lookup in the mapping, similar to what the pkix go lib does, so now I'm starting to think I actually am missing a big piece (that mapping). 

I am going to do some playing around with the other PR as I now think it might be best to build on that and try to extend it and we will need to add explicit fields (givenname, domaincomponent, etc), which I was trying really hard to avoid.